apps/prairielearn/src/lib/workspace.ts

```diff
@@ -444,7 +448,7 @@ async function initialize(workspace_id: string): Promise<InitializeResult> {
         }
         try {
           // Discard names with directory traversal outside the home directory
-          if (!contains(remotePath, path.join(remotePath, file.name), false)) {
+          if (!contains(sourcePath, path.join(sourcePath, file.name), false)) {
```

Comment:
Bugfix: `contains` requires the parent directory to be an absolute path to work as expected.